Code/GraphMol/ChemReactions/DaylightParser.cpp

```diff
@@ -87,11 +87,39 @@ std::vector<std::string> splitSmartsIntoComponents(
   }
   return res;
 }
+
+ROMol* constructMolFromString(const std::string &txt,
+    std::map<std::string,std::string> *replacements, bool useSmiles, bool sanitize) {
+  ROMol *mol;
+  if(!useSmiles){
+    mol=SmartsToMol(txt,0,false,replacements);
+    if(sanitize){
+      try{
+        mol->updatePropertyCache();
+        RingInfo *ringInfo=mol->getRingInfo();
+        if(!ringInfo->isInitialized()){
+          ringInfo->initialize();
+        }
+      }
+      catch (const MolSanitizeException &){
+        std::cerr << "Molecule could not be sanitized" << std::endl;
+      }
+    }
+  }
+  else{
+    mol=SmilesToMol(txt,0,sanitize,replacements);
+  }
+  return mol;
+}
+
+
+
+
 }  // end of namespace DaylightParserUtils
 
 ChemicalReaction *RxnSmartsToChemicalReaction(
     const std::string &text, std::map<std::string, std::string> *replacements,
-    bool useSmiles) {
+    bool useSmiles, bool sanitize) {
```

Comment:
So you mean sanitizing the molecules of a reaction always when useSmiles is set to True and otherwise ignoring sanitization? I added this function to improve the drawing of reactions, not sanitizing them leads to ugly images.